src/services/myMap.js

```diff
@@ -0,0 +1,140 @@
+const { promisify } = require('util');
+
+const { sale: SALE } = require('../config');
+
+// eslint-disable-next-line no-extend-native
+Array.prototype.myMap = async (oldArray, callback) => {
+  const newArray = [];
+
+  for (let i = 0; i < oldArray.length; i++) {
+    // eslint-disable-next-line no-await-in-loop
+    newArray.push(await callback(oldArray[i], i, oldArray));
+  }
+
+  return newArray;
+};
+
+function randomNumber(min, max) {
+  return Math.floor(Math.random() * (max - min + 1)) + min;
+}
+
+async function callbackFunction(callback) {
+  const promise = promisify((time, func) => setTimeout(func, time));
+
+  await promise(SALE.TIME_GENERATE_SALE);
+
+  const number = randomNumber(SALE.MIN, SALE.MAX);
+
+  if (number >= SALE.NOT_MORE_THAN) {
+    return callback('Incorrect sale!');
+  }
+
+  return callback(0, number);
```

Comment:
Usually, `null` is used instead of `0` when there is no error.
`callback(null, result);`